packages/data/src/items/actions.js

```diff
@@ -79,3 +79,22 @@ export function* updateProductStock( product, quantity ) {
 		return false;
 	}
 }
+
+export function* createProductTemplate( itemFields, query ) {
+	try {
+		const url = addQueryArgs(
+			`${ NAMESPACE }/products/templates`,
+			query || {}
+		);
+		const newItem = yield apiFetch( {
+			path: url,
+			method: 'POST',
+			data: itemFields,
+		} );
+		yield setItem( 'products', newItem.id, newItem );
+		return newItem;
+	} catch ( error ) {
+		yield setError( 'createItem', error );
```

Comment:
Is this too generic an error name? Should this be `createProductFromTemplate` or similar?